core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java

```diff
@@ -0,0 +1,96 @@
+import org.eclipse.edc.identityhub.transform.JsonObjectToPresentationQueryTransformer;
+import org.eclipse.edc.identityservice.api.v1.PresentationApiController;
+import org.eclipse.edc.identityservice.api.validation.PresentationQueryValidator;
+import org.eclipse.edc.jsonld.spi.JsonLd;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.transform.spi.TypeTransformerRegistry;
+import org.eclipse.edc.validator.spi.JsonObjectValidatorRegistry;
+import org.eclipse.edc.web.jersey.jsonld.JerseyJsonLdInterceptor;
+import org.eclipse.edc.web.jersey.jsonld.ObjectMapperProvider;
+import org.eclipse.edc.web.spi.WebService;
+
+import java.net.URISyntaxException;
+
+import static org.eclipse.edc.spi.CoreConstants.JSON_LD;
+
+public class PresentationApiExtension implements ServiceExtension {
+
+    public static final String RESOLUTION_SCOPE = "resolution-scope";
+    public static final String RESOLUTION_CONTEXT = "resolution";
+    @Inject
+    private TypeTransformerRegistry typeTransformer;
+
+    @Inject
+    private JsonObjectValidatorRegistry validatorRegistry;
+
+    @Inject
+    private WebService webService;
+
+    @Inject
+    private AccessTokenVerifier accessTokenVerifier;
+
+    @Inject
+    private CredentialQueryResolver credentialResolver;
+
+    @Inject
+    private PresentationGenerator presentationGenerator;
+
+    @Inject
+    private JsonLd jsonLd;
+
+    @Inject
+    private TypeManager typeManager;
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        // setup validator
+        validatorRegistry.register(PresentationQuery.PRESENTATION_QUERY_TYPE_PROPERTY, new PresentationQueryValidator());
+
+
+        // Setup API
+        cacheContextDocuments();
+        var controller = new PresentationApiController(validatorRegistry, typeTransformer, credentialResolver, accessTokenVerifier, presentationGenerator, context.getMonitor());
+
+        var jsonLdMapper = typeManager.getMapper(JSON_LD);
+        webService.registerResource(RESOLUTION_CONTEXT, new ObjectMapperProvider(jsonLdMapper));
+        webService.registerResource(RESOLUTION_CONTEXT, new JerseyJsonLdInterceptor(jsonLd, jsonLdMapper, RESOLUTION_SCOPE));
+        webService.registerResource(RESOLUTION_CONTEXT, controller);
+
+        // register transformer
+        typeTransformer.register(new JsonObjectToPresentationQueryTransformer(jsonLdMapper));
+        typeTransformer.register(new JsonValueToGenericTypeTransformer(jsonLdMapper));
+    }
+
+    private void cacheContextDocuments() {
+        try {
+            jsonLd.registerCachedDocument("https://identity.foundation/presentation-exchange/submission/v1", Thread.currentThread().getContextClassLoader().getResource("presentation-exchange.v1.json").toURI());
```

Comment:
We should use the classloader of the class, not the TCCL as that may be defined differently in different contexts

Also, can we make these constants and introduce a var for the classloader?

Reply:
yeah, this was copy-pasted from test code :/